src/Form/TypeGuesser/CronExpressionTypeGuesser.php

```diff
@@ -4,18 +4,23 @@
 
 namespace Setono\CronExpressionBundle\Form\TypeGuesser;
 
-use Brick\Reflection\ReflectionTools;
 use Cron\CronExpression;
-use ReflectionClass;
-use ReflectionException;
 use Setono\CronExpressionBundle\Form\Type\CronExpressionType;
 use Symfony\Component\Form\FormTypeGuesserInterface;
 use Symfony\Component\Form\Guess\Guess;
 use Symfony\Component\Form\Guess\TypeGuess;
 use Symfony\Component\Form\Guess\ValueGuess;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
 
 final class CronExpressionTypeGuesser implements FormTypeGuesserInterface
 {
+    protected PropertyTypeExtractorInterface $extractor;
+
+    public function __construct(PropertyTypeExtractorInterface $extractor)
```

Comment:
The problem with this is that it's a BC break

Reply:
the class is final, so it shouldn't be that much of a problem

(also should i make the member private instead when the class can't be overwritten anyway?)

Reply:
Yes, I agree, but nevertheless it's a BC break, and I try to keep that :| Do you think you'll be able to create the compatibility layer?

> (also should i make the member private instead when the class can't be overwritten anyway?)
Yes  :)

Reply:
i added phpDocumentor as compatibility layer

what i also could do is set the TypeExtractor via a setter function instead of constructor ... would that have been okay too?

Reply:
See this: https://github.com/Setono/CronExpressionBundle/pull/40#discussion_r1424078959